### PR TITLE
fix(DavClient): Provide proper feedback for dav requests

### PR DESCRIPTION
**Pull request by Fenn-CS**
Before this commit there is no error handling whatsoever for this requests and the user won't get any visual feedback whatsoever irrespective of what goes down at the server level.

Resolves : https://github.com/nextcloud-gmbh/customer-feature-requests/issues/796

| Before | After |
| ----------| -------- |
|Screenshot from 2024-07-02 16-47-17 |Screenshot from 2024-07-02 16-21-24|

### TODO

- [x] Use Nextcloud logger?
- [ ] Toast for success necessary?
- [x] Before and after demo

### CODE REVIEW

File: src/components/SharingSidebarView.vue [hunk 1]
```diff
@@ -374,9 +377,17 @@ export default {
...
+			// TODO: Block UI during save
+			try {
+				await client.propPatch(this.model, this.list.filter(rule => !rule.inherited))
+				logger.debug('Permissions updated successfully')
+			} catch (error) {
+				logger.error('Failed to save changes:', { error })
+				Vue.set(this.list, index, itemRestorePoint)
+				showError(error)
+			} finally {
+				// TODO: Unblock the UI after the operation completes
```
Comment (artonge) [inline src/components/SharingSidebarView.vue:389]: What does it mean?

File: src/components/SharingSidebarView.vue [hunk 2]
```diff
@@ -185,6 +186,7 @@ import NcSelect from '@nextcloud/vue/dist/Components/NcSelect.js'
 import Tooltip from '@nextcloud/vue/dist/Directives/Tooltip.js'
 import Plus from 'vue-material-design-icons/Plus.vue'
 import Close from 'vue-material-design-icons/Close.vue'
+import logger from '../services/logger.ts'
```
Comment (skjnldsv) [inline src/components/SharingSidebarView.vue:189]: Can we sort the import block. :see_no_evil:

File: src/client.js [hunk 3]
```diff
@@ -252,12 +253,30 @@ class AclDavService {
 	propPatch(model, acls) {
 		const aclList = []
 		for (const i in acls) {
-			aclList.push({ type: ACL_PROPERTIES.PROPERTY_ACL_ENTRY, data: acls[i].getProperties() })
+		  aclList.push({ type: ACL_PROPERTIES.PROPERTY_ACL_ENTRY, data: acls[i].getProperties() })
```
Comment (skjnldsv) [inline src/client.js:256]:
```suggestion
			aclList.push({ type: ACL_PROPERTIES.PROPERTY_ACL_ENTRY, data: acls[i].getProperties() })
```

File: src/components/SharingSidebarView.vue (hunk 2 above)
Comment (Fenn-CS) [reply src/components/SharingSidebarView.vue:185]: Use `Volar` sorting here now

File: src/components/SharingSidebarView.vue (hunk 1 above)
Comment (artonge) [reply src/components/SharingSidebarView.vue:389]: Or remove the comments

**Review by skjnldsv — APPROVED** [review]
Please fix the mixed space-tabs indents

**Review by artonge — APPROVED** [review]

**Comment by Fenn-CS** [comment]
Cypress tests failures unrelated.

**Pull request merged**